lms/djangoapps/gating/api.py

```diff
@@ -63,35 +61,60 @@ def _get_subsection_percentage(subsection_grade):
     """
     Returns the percentage value of the given subsection_grade.
     """
-    if subsection_grade.graded_total.possible:
-        return float(subsection_grade.graded_total.earned) / float(subsection_grade.graded_total.possible) * 100.0
-    else:
-        return 0
+    return _calculate_percentage(subsection_grade.graded_total.earned, subsection_grade.graded_total.possible) * 100.0
 
 
-def evaluate_entrance_exam(course, subsection_grade, user):
+def _calculate_percentage(earned, possible):
+    """
+    Returns the percentage of the given earned and possible values.
+    """
+    return float(earned) / float(possible) if possible else 0.0
```

Comment:
Terminology nit: I think of "percent" as being a number in the range [0, 100], so it's kind of weird to have this method be named `_calculate_percentage` and return a number in the range [0, 1] that the caller must multiply by 100.

I'd prefer to see this method do the `return_value * 100` calculation itself, or be renamed `_calculate_ratio`.